### DISCUSSION TITLE
Setting git branch to point to the tip using GitHub ui

### DISCUSSION THREAD

**Discussion by davidsalazar** [community/community#143794] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

The other day I was on GitHub web ui and I accidentally set one of the branches (origin/develop) to point to an older commit. I have no idea how I did it. 

How can I make it point back to the tip of that branch again using the GitHub web ui?

**Comment by mariomartv** [comment]
In the GitHub web UI, you can't directly change a branch's commit reference (or "reset" it) like you would in a command line with git reset. However, you can bring the branch back to its latest state by creating a pull request (PR) that points it to the most recent commit on the branch. Here’s how:

Create a Temporary Branch:

Go to your repository on GitHub.

Switch to the branch that’s currently at the correct, most recent commit (e.g., origin/main or whatever branch is up to date).

Create a new branch from this state by selecting "New branch" and naming it something like temp-recover.

Create a Pull Request:

Once you have this temporary branch, go back to the origin/develop branch (or the branch you accidentally changed).

Click on Pull requests in the top menu, and create a new pull request from temp-recover to origin/develop.

Ensure the PR contains all the changes you need to bring origin/develop back to its latest state.

Merge the Pull Request:

Review the pull request to make sure all changes are correct.

Click Merge pull request to apply it, which will update the origin/develop branch to the latest commit in the temporary branch.

Delete the Temporary Branch:

After merging, you can delete the temp-recover branch if it was just used for this purpose.

This process effectively "resets" your branch to the latest state without using Git on the command line. In the future, to prevent accidentally resetting branches, you could consider enforcing branch protection rules under Settings > Branches.

**Comment by mastertanzeel** [comment]
Hey David,

If you accidentally set a branch (like `origin/develop`) to point to an older commit in GitHub and want to reset it to point to the latest commit, here’s how to fix it:

### Option 1: Use the GitHub Web UI (if there's a recent commit to reference)
1. **Go to the Commits page for the branch:**
   - Navigate to your repository on GitHub.
   - Click on **"Branches"** to see all branches or go directly to the `develop` branch.
   - Click on the **"Commits"** tab to view the commit history of `origin/develop`.

2. **Find the latest commit:**
   - Identify the commit hash of the latest or desired commit you want `origin/develop` to point to.

3. **Create a new branch or force-update the branch:**
   Unfortunately, GitHub doesn’t directly allow you to “force update” a branch from the web UI, so you’ll need to follow these steps locally or use a workaround (Option 2).

### Option 2: Use the Git Command Line (most straightforward option)
If the above approach isn't possible directly from the web UI, you can use the following Git commands on your local machine:

1. **Fetch the latest changes:**
   ```bash
   git fetch origin
   ```

2. **Reset `develop` to the latest commit:**
   ```bash
   git checkout develop
   git reset --hard origin/develop
   ```

3. **Push the reset branch back to GitHub:**
   ```bash
   git push --force origin develop
   ```

   This will update `origin/develop` on GitHub to point to the latest commit.

### Option 3: Use a Pull Request (Web UI workaround)
If you're limited to using only the GitHub Web UI and don't have direct access to the command line:

1. **Create a new branch** off the correct `develop` tip if it’s accessible in the **Commits** history.
2. **Make a Pull Request (PR)** from this new branch into the `develop` branch to update it.
3. **Merge the PR** into `develop` to update it to the latest commit.

This approach doesn’t reset the branch directly but achieves the same result by merging the latest changes into it. 

### Important Note
To avoid this issue in the future, ensure you have branch protection rules enabled (if you’re an admin on the repository) to prevent force-pushing to critical branches like `develop`. This way, accidental resets are minimized.
